friends/src/components/FriendList.js

```diff
@@ -0,0 +1,113 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import { Container, Form, Button, Label, Input, Alert } from "reactstrap";
+import Friend from "./Friend";
+
+import axios from "axios";
+class FriendList extends React.Component {
+  state = {
+    friends: [],
+    nameInput: "",
+    ageInput: "",
+    emailInput: "",
+    error: ""
+  };
+
+  componentDidMount() {
+    axios
+      .get("http://localhost:5000/friends")
+      .then(res => this.setState({ friends: res.data }))
+      .catch(err => console.log(err));
```

Comment:
I would call your friends `API` inside of the `CDM` in `App.js`. The reason why, is because the way you currently have it set up only children of `FriendList` will be able get the `friends` state and `methods` but, no other component. The reason why this is bad, is because if any other component needs access to your `friends` state or methods besides children components of `FriendList` they won't be able to get it. Putting it in app will allow every single component to use that state and or methods in `App.js` simply because every component is a child of `App`.